frontend/README.md

```diff
@@ -0,0 +1,162 @@
+# Translate.Next — a better Translate app for Pontoon
+
+
+## Tools
+
+<table>
+    <tr>
+        <td>Bootstrapper</td>
+        <td>create-react-app — https://github.com/facebook/create-react-app</td>
+    </tr>
+    <tr>
+        <td>Package manager</td>
+        <td>Yarn — https://yarnpkg.com/</td>
+    </tr>
+    <tr>
+        <td>Views</td>
+        <td>React — https://reactjs.org/</td>
+    </tr>
+    <tr>
+        <td>Data</td>
+        <td>Redux — https://redux.js.org/</td>
+    </tr>
+    <tr>
+        <td>Tests</td>
+        <td>Jest — http://jestjs.io/</td>
+    </tr>
+    <tr>
+        <td>Type checking</td>
+        <td>Flow — https://flow.org/</td>
+    </tr>
+    <tr>
+        <td>Localization</td>
+        <td>Fluent — https://projectfluent.org/</td>
+    </tr>
+    <tr>
+        <td>Style</td>
+        <td>CSS</td>
+    </tr>
+</table>
+
+
+## Code architecture
+
+### Where code goes
+
+`src/core/` contains features that are shared in the application, in the form of modules. There should be as little code as possible in this folder.
+
+`src/modules/` contains the self-contained features of the application, separated in modules.
+
+`src/rootReducer.js` creates the main reducer to be used with Redux. When adding a new module with a reducer, make sure to include that reducer to `rootReducer`.
+
+### Modules
+
+Each module should be in a folder and have an `index.js` file that serves as the module's public interface. Outside of the module, always import from the module's index, and never from specific files. This allows us to easily evolve modules and keep things decoupled, which reduces code complexity.
+
+Here are the files commonly found in a module:
+
+- `index.js` — public interface of the module
+- `actions.js` — actions that can be used to fetch data from an API, trigger changes, etc.
+- `reducer.js` — a single Redux reducer (if you need to have more than one reducer, you probably actually need to make several modules)
+- `constants.js` — a list of constants required by the module or other modules. It is recommended to expose a `NAME` constant here, that will contain the key to use to expose the reducer in the global store.
```

Comment:
Perhaps change the wording to make this a bit more clear:

> ... It's recommended to define a `NAME` constant here which should be a unique identifier of the module. This name will be used in `combineReducers` and in all `mapStateToProps` to identify the subtree of the global state relevant to this module.

Reply:
Also, having a separate file when there's only one constant might be a bit of an overkill. There two modules in this PS and both only have `NAME` defined. I'd put it directly in `index.js` and then figure out the good practice for defining more once they are actually needed.